packages/xstate-graph/src/types.ts

```diff
@@ -162,7 +162,10 @@ export interface TraversalOptions<TState, TEvent extends EventObject>
     vctx: VisitedContext<TState, TEvent>
   ) => boolean;
   filter?: (state: TState, event: TEvent) => boolean;
-  getEvents?: (state: TState, cases: EventCaseMap<TState, TEvent>) => TEvent[];
+  getEvents?: (
+    state: TState,
+    cases: EventCaseMap<TState, TEvent>
+  ) => ReadonlyArray<TEvent>;
```

Comment:
using `ReadonlyArray` is often better than using regular arrays in types because it accepts both readonly arrays and mutable arrays. In our context it's super convenient to use `[] as const` to narrow contained strings etc to literals and that produces readonly arrays